File: yamcs-simulation/yamcs-studio-workspace/BOY Examples/scripts/ExecuteScriptFromAction.js
importPackage(Packages.org.csstudio.opibuilder.scriptUtil);
importPackage(Packages.org.eclipse.jface.dialogs);
importPackage(Packages.java.lang);

var color;
var colorName;
if(Math.random()>0.5){
	color = ColorFontUtil.getColorFromRGB(0,160,0);
	colorName = "green";
}
else{
	color = ColorFontUtil.RED;
	colorName = "red";
}
display.getWidget("myIndicator").setPropertyValue("background_color", color);
widget.setPropertyValue("foreground_color", color);
	
MessageDialog.openInformation(
			null, "Dialog from JavaScript", "JavaScript says: My color is " + colorName);

File: yamcs-simulation/yamcs-studio-workspace/BOY Examples/scripts/ExecuteScriptFromAction.py
from org.csstudio.opibuilder.scriptUtil import PVUtil
from org.csstudio.opibuilder.scriptUtil import ColorFontUtil
from org.eclipse.jface.dialogs import MessageDialog
from java.lang import Math



if Math.random() > 0.5:
	color = ColorFontUtil.getColorFromRGB(0,160,0)
	colorName = "green"

else:
	color = ColorFontUtil.RED
	colorName = "red"

import WidgetUtil
WidgetUtil.setBackColor(display, "myIndicator", color)
WidgetUtil.setMyForeColor(widget, color)

MessageDialog.openInformation(
			None, "Dialog from Python Script", "Python says: my color is " + colorName);

